benchmarks/time/src/main/scala/strawman/collection/immutable/ChampHashSetBenchmark.scala

```diff
@@ -0,0 +1,158 @@
+package strawman.collection.immutable
+
+import java.util.concurrent.TimeUnit
+
+import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra.Blackhole
+
+@BenchmarkMode(scala.Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Fork(1)
+@Warmup(iterations = 8)
+@Measurement(iterations = 8)
+@State(Scope.Benchmark)
+class ChampHashSetBenchmark {
+  @Param(scala.Array("0", "1", "2", "3", "4", "7", "8", "15", "16", "17", "39", "282", "4096", "131070", "7312102"))
+  var size: Int = _
+
+  var xs: ChampHashSet[Long] = _
+  var zs: ChampHashSet[Long] = _
+  var zipped: ChampHashSet[(Long, Long)] = _
+  var randomIndices: scala.Array[Int] = _
+  def fresh(n: Int) = ChampHashSet((1 to n).map(_.toLong): _*)
```

Comment:
It might be worthwhile varying the key types during the benchmark to prevent JIT from having an unrealistically easy time inlining the hashing/equality (for `Double` keys, that goes through `BoxesRuntime.{equals,hashCode}`).

This is a general problem for our collection benchmarks, but it is particularly important for hash-structures IMO.

Reply:
This is what I tried in the benchmark of the existing Scala hash map: https://github.com/scala/scala/blob/5c93cd2431276fe3c712cb60e8a7a696c1776f10/test/benchmarks/src/main/scala/scala/collection/mutable/HashMapBenchmark.scala#L25-L33